### PR TITLE
Remove disaggregated mode without s3

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #8276

Problem Summary:

### What is changed and how it works?
- remove `flash.use_autoscaler_without_s3` https://github.com/pingcap/tiflash/pull/7116
- remove disaggregated mode without s3 https://github.com/pingcap/tiflash/pull/6248

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/tests/gtest_storage_disaggregated.cpp [hunk 1]
```diff
@@ -1,123 +0,0 @@
...
-    ASSERT_EQ(getProxyLabelByDisaggregatedMode(DisaggregatedMode::Compute), "tiflash_compute");
-    ASSERT_EQ(getProxyLabelByDisaggregatedMode(DisaggregatedMode::Compute), DISAGGREGATED_MODE_COMPUTE_PROXY_LABEL);
-
-    ASSERT_EQ(getProxyLabelByDisaggregatedMode(DisaggregatedMode::Storage), "tiflash");
-    ASSERT_EQ(getProxyLabelByDisaggregatedMode(DisaggregatedMode::Storage), DEF_PROXY_LABEL);
-
-    ASSERT_EQ(getProxyLabelByDisaggregatedMode(DisaggregatedMode::None), "tiflash");
-    ASSERT_EQ(getProxyLabelByDisaggregatedMode(DisaggregatedMode::None), DEF_PROXY_LABEL);
-}
-CATCH
```
Comment (JaySon-Huang) [inline dbms/src/Flash/tests/gtest_storage_disaggregated.cpp:120]: Keep this test?

Comment (SeaRise) [reply dbms/src/Flash/tests/gtest_storage_disaggregated.cpp]: Okay, added back

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**